core/dbt/contracts/util.py

```diff
@@ -260,6 +259,12 @@ def upgrade_seed_content(node_content):
     ):
         if attr_name in node_content:
             del node_content[attr_name]
+        # In v1.4, we switched SeedNode.depends_on from DependsOn to MacroDependsOn
+        # I'm not actually sure if this logic is necessary?
+        # TK: actual automated testing for these methods
+        # https://github.com/dbt-labs/dbt-core/issues/6753
+        if "depends_on" in node_content and "nodes" in node_content["depends_on"].keys():
+            del node_content["depends_on"]["nodes"]
```

Comment:
I tested both with & without this change:
- Creating a seed with a pre/post hook
- Generating a manifest using v1.3
- Switching to this branch
- Using that v1.3 manifest for `--state` comparison

If someone had a hook calling `ref()` in v1.3, they will see the explicit error message on upgrading. I'm okay with that, I'm saying it's an anti-pattern & unsupported.

Reply:
I agree that it does make the change more explicit.

Reply:
Minor readability suggestion, lines 262/3 are equivalent to this and require less back and forth to read:
```
node_content.get("depends_on", {}).pop("nodes", None)
```

If the goal is to be explicit about checking for "depends_on", then I would write it this way:
```
if "depends_on" in node_content:
    node_content["depends_on"].pop("nodes", None)
```

Reply:
Made this change for simplicity / readability. Performed additional manual testing, since we still don't have automated testing for seeds in this manifest upgrade flow (tracked in https://github.com/dbt-labs/dbt-core/issues/6753)